dispatch.go

```diff
@@ -1,13 +1,24 @@
 package rpc
 
-import "sync"
+import (
+	"errors"
+	"sync"
+)
 
 type DecodeNext func(interface{}) error
-type ServeHook func(DecodeNext) (interface{}, error)
-type ServeNotifyHook func(DecodeNext) error
+
+type ServeHookDescription struct {
+	MakeArg  func() interface{}
+	Callback func(arg interface{}) (ret interface{}, err error)
```

Comment:
I never got to reply to this suggestion --- I actually don't like "callback", I think it means something else, not this.  I think of it as a function Y you pass into function X, that function X calls when it's done.  That's not this.

I think the right word is `Handler`